fvm/evm/offchain/blocks/blocks.go

```diff
@@ -0,0 +1,122 @@
+package blocks
+
+import (
+	"fmt"
+
+	gethCommon "github.com/onflow/go-ethereum/common"
+
+	"github.com/onflow/flow-go/fvm/evm/handler"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const BlockStoreLatestBlockMetaKey = "LatestBlockMeta"
+
+// Blocks facilitates access to the recent block hash values
+// and also the latest executed block meta data
+type Blocks struct {
+	chainID     flow.ChainID
+	storage     types.BackendStorage
+	rootAddress flow.Address
+	bhl         *handler.BlockHashList
+}
+
+// NewBlocks constructs a new blocks type
+func NewBlocks(
+	chainID flow.ChainID,
+	rootAddress flow.Address,
+	storage types.BackendStorage,
+) (*Blocks, error) {
+	var err error
+	blocks := &Blocks{
+		chainID:     chainID,
+		storage:     storage,
+		rootAddress: rootAddress,
+	}
+	blocks.bhl, err = handler.NewBlockHashList(
+		storage,
+		rootAddress,
+		handler.BlockHashListCapacity,
+	)
+	if err != nil {
+		return nil, err
+	}
+	// if empty insert genesis block hash
+	if blocks.bhl.IsEmpty() {
+		genesis := types.GenesisBlock(chainID)
+		err = blocks.PushBlockMeta(
+			NewMeta(
+				genesis.Height,
+				genesis.Timestamp,
+				genesis.PrevRandao,
+			))
+		if err != nil {
+			return nil, err
+		}
+		// push block hash
+		err = blocks.PushBlockHash(
+			genesis.Height,
+			types.GenesisBlockHash(chainID))
```

Comment:
I'm afraid this won't work properly for `testnet`, because the `Genesis` block there had no `PrevRandao` field, and the hash might be different to what is stored on the EVM GW, and what was emitted within the `EVM.BlockExecuted` event.

Reply:
I'm updating the GenesisBlockHash to return the proper value for testnet genesis, what was the correct value that we used the first time ?